### ISSUE TITLE
Load/Unload issue depending on naming of module file

### ISSUE THREAD

**Issue by dr-br** [cmd-ntrf/jupyter-lmod#47]
Depending on the naming of the Lmod module file, the module is listed under "LOADED MODULES" or not. When loading the module via the "Load" button, it gets loaded. However, as it does not appear in the "LOADED MODULES" list, it can't be unloaded.

Example:
With the module file `9-test.lua` everything works as expected, with `9.lua` it does not. Both files are a symlink to exactly the same module file.

As in #26 we already identified a regex as the root of all evil, I would eventually point to lines 15+16 in this file: https://github.com/cmd-ntrf/jupyter-lmod/blob/main/lmod/__init__.py

Thanks in advance :)

**Comment by dr-br** [comment]
```
(^[\w\/]+\/[\w.]+)
```

**Comment by dr-br** [comment]
```
^(.*?\/[^.][^\/]*)$
```

**Comment by cmd-ntrf** [comment]
Can you provide an example with the complete path for the module? Or do you have module with just a number as the name at the root of the module path?

**Comment by cmd-ntrf** [comment]
I am currently unable to reproduce.

I have this barebone module structure:
```
modules tree
.
├── jupyter
│   └── minimal
│       └── 2.lua
└── rstudio-server
    └── 1.4.43.lua
```

<img width="283" alt="image" src="https://user-images.githubusercontent.com/884095/153635627-5ef495b9-6dec-4e26-b5f6-7adaa47c8d95.png">

`jupyter/minimal/2` shows up as expected.

**Comment by cmd-ntrf** [comment]
The issue is with `MODULE_REGEX_NO_HIDDEN` which cannot match `jupyter/minimal/2`, leading jupyter-lmod to think wrongfully that it is an hidden module.

**Comment by cmd-ntrf** [comment]
@dr-br : can you try branch hidden_regex and confirm it solves your issue.

**Comment by dr-br** [comment]
Yes, but now there is another issue:  

![grafik](https://user-images.githubusercontent.com/45423137/153648036-02d96b8f-ebbc-4d5d-b599-a83d02e9fbed.png)

As you can see, my openfoam module loads among others `mpi/openmpi/4.0` (right side, `module list`), but is only displayed as `mpi/openmpi` on the left hand side. I can unload this module with the button, (`module list` in terminal is empty) but it remains visible within the "LOADED MODULES" listing :(

**Comment by cmd-ntrf** [comment]
After loading openfoam with the command-line, could you echo the content of `$LOADEDMODULES` and paste it here:
``` 
echo $LOADEDMODULES
```

**Comment by cmd-ntrf** [comment]
I have made a few improvements to the PR. I am not sure why the version of the `mpi/openmpi/4.0` module would be cut and I have not been able to reproduce locally.

**Comment by dr-br** [comment]
Update:
We located the issue with diverging `module li` and `$LOADEDMODULES`. So from this point of view, this ticket could be closed, the regex-fix and your other modifications solve this issue.
However, having a look at [1], it seems that the variable `$LOADEDMODULES` is only for compatibility reasons with Tmod. It would be safer to eventually use `module -t list` instead of `$LOADEDMODULES`. In our specific case, we could not rely on this variable.

So instead of `loadedmodules = os.environ.get('LOADEDMODULES', '')` grab the output of `module -t list`.

[1]: https://lmod.readthedocs.io/en/latest/160_debugging_modulefiles.html?highlight=loadedmodules#lua-modulefiles

**Comment by dr-br** [comment]
Thank YOU! Works like a charm.

**Comment by dr-br** [comment]
I think the version string in setup.py still needs to be updated.

**Comment by cmd-ntrf** [comment]
Released again as 2.0.5
